controller/workitem_whitebox_test.go

```diff
@@ -295,6 +348,34 @@ func (rest *TestWorkItemREST) TestConvertWorkItem() {
 		assert.Equal(t, "title", wi2.Attributes[workitem.SystemTitle])
 		assert.Nil(t, wi2.Attributes[workitem.SystemDescription])
 	})
+	// TODO (ibrahim) - This test should be removed once field name migration is completed
+	rest.T().Run("check field names", func(t *testing.T) {
+		request := &http.Request{Host: "localhost"}
+		wi := workitem.WorkItem{
+			Type:    fxt.WorkItemTypes[0].ID,
+			SpaceID: fxt.Spaces[0].ID,
+			Fields: map[string]interface{}{
+				workitem.SystemTitle:       "title",
+				workitem.SystemDescription: "xyz",
+			},
+		}
+		convertedWI, err := ConvertWorkItem(request, *fxt.WorkItemTypes[0], wi)
+		require.NoError(t, err)
+		for attr := range convertedWI.Attributes {
+			oldFieldName, ok := workitem.NewToOldFieldNameMap[attr]
+			if ok {
+				// Check old and new field have the same value
+				assert.Equal(t, convertedWI.Attributes[attr], convertedWI.Attributes[oldFieldName])
```

Comment:
I wonder if the `convertedWI.Attributes[oldFieldName]` shouldn't be the expected value. I mean you're adding `system_title` and `system.title` already existed. Shouldn't the value of `system.title` be the expected one?

Reply:
Yes. This check if to make sure we have both the field names. Like `check if system.title and system_title` are present`. The following lines, line 372 and line 373 check if the values are correct.
```go
oldFieldName, ok := workitem.NewToOldFieldNameMap[attr]
if ok {
	// Check old and new field have the same value
	assert.Equal(t, convertedWI.Attributes[attr], convertedWI.Attributes[oldFieldName])
	// If the payload had a value, both fields should have this value
	val, ok := wi.Fields[attr]
	if ok {
		assert.Equal(t, convertedWI.Attributes[attr], val)
		assert.Equal(t, convertedWI.Attributes[oldFieldName], val)
	}
}
```

Reply:
All I had in mind was replacing this:

```go
assert.Equal(t, convertedWI.Attributes[attr], convertedWI.Attributes[oldFieldName])
```

with 

```go
assert.Equal(t, convertedWI.Attributes[oldFieldName], convertedWI.Attributes[attr])
```

The expected value is the one of the old field name (e.g. `system.title`).

Reply:
Thanks. Done in https://github.com/fabric8-services/fabric8-wit/pull/2333/commits/9f1d82daac536c863dc99157c6a3f184873c6f97